irc.py

```diff
@@ -149,13 +161,20 @@ def _send_chan_info(self, channel_name: bytes, slchan: slack.Channel):
                     continue
                 name = u.name.encode('utf8')
                 prefix = b'@' if u.is_admin else b''
-                userlist.append(prefix + name)
-
-            users = b' '.join(userlist)
+                replystr = b' '.join(reply)
+                if (len(replystr) + len(prefix + name + b' ') + overhead) > MAX_LINE_LEN:
+                    replies.append(replystr)
+                    reply = []
+                reply.append(prefix + name)
+            if reply:
+                replies.append(b' '.join(reply))
+
+        if replies:
+            for line in replies:
+                self._sendreply(Replies.RPL_NAMREPLY, line, ['=', channel_name])
```

Comment:
can't you just do all this without calculating the overhead, sending 1 per line instead? I think it would be much more simple. Ok you do have more system calls, but I don't think it would hit performances in a noticeable way anyway.

Reply:
Yeah it would be quite inefficient but would presumably work. I will have to check how much slower it is and how often NAMES is called.